src/controller/artifact/processor/default.go

```diff
@@ -16,43 +16,87 @@ package processor
+	"github.com/goharbor/harbor/src/pkg/artifact"
+	"github.com/goharbor/harbor/src/pkg/registry"
 )
 
 // ArtifactTypeUnknown defines the type for the unknown artifacts
 const ArtifactTypeUnknown = "UNKNOWN"
 
 var (
+	// DefaultProcessor is to process artifact which has no specific processor
+	DefaultProcessor = &defaultProcessor{RegCli: registry.Cli}
+
 	artifactTypeRegExp = regexp.MustCompile(`^application/vnd\.[^.]*\.(.*)\.config\.[^.]*\+json$`)
 )
 
 // the default processor to process artifact
-// currently, it only tries to parse the artifact type from media type
 type defaultProcessor struct {
-	mediaType string
+	RegCli registry.Client
 }
 
-func (d *defaultProcessor) GetArtifactType() string {
+func (d *defaultProcessor) GetArtifactType(ctx context.Context, artifact *artifact.Artifact) string {
 	// try to parse the type from the media type
-	strs := artifactTypeRegExp.FindStringSubmatch(d.mediaType)
+	strs := artifactTypeRegExp.FindStringSubmatch(artifact.MediaType)
 	if len(strs) == 2 {
 		return strings.ToUpper(strs[1])
 	}
 	// can not get the artifact type from the media type, return unknown
 	return ArtifactTypeUnknown
 }
-func (d *defaultProcessor) ListAdditionTypes() []string {
+func (d *defaultProcessor) ListAdditionTypes(ctx context.Context, artifact *artifact.Artifact) []string {
 	return nil
 }
-func (d *defaultProcessor) AbstractMetadata(ctx context.Context, manifest []byte, artifact *artifact.Artifact) error {
-	// do nothing currently
-	// we can extend this function to abstract the metadata in the future if needed
+
+// The default processor will process user-defined artifact.
+// AbstractMetadata will abstract data in a specific way.
+// Annotation keys in artifact annotation will decide which content will be processed in artifact.
+// Here is a manifest example:
+// {
+//   "schemaVersion": 2,
+//   "config": {
+//       "mediaType": "application/vnd.caicloud.model.config.v1alpha1+json",
+//       "digest": "sha256:be948daf0e22f264ea70b713ea0db35050ae659c185706aa2fad74834455fe8c",
+//       "size": 187,
+//       "annotations": {
+//           "io.goharbor.artifact.v1alpha1.skip-list": "metrics,git"
+//       }
+//   },
+//   "layers": [
+//       {
+//           "mediaType": "image/png",
+//           "digest": "sha256:d923b93eadde0af5c639a972710a4d919066aba5d0dfbf4b9385099f70272da0",
+//           "size": 166015,
+//           "annotations": {
+//               "io.goharbor.artifact.v1alpha1.icon": ""
+//           }
+//       },
+//       {
+//           "mediaType": "application/tar+gzip",
+//           "digest": "sha256:d923b93eadde0af5c639a972710a4d919066aba5d0dfbf4b9385099f70272da0",
+//           "size": 166015
+//       }
+//   ]
+// }
+func (d *defaultProcessor) AbstractMetadata(ctx context.Context, artifact *artifact.Artifact, manifest []byte) error {
+	annotationParser := annotation.NewParser()
+	err := annotationParser.Parse(ctx, artifact, manifest)
+	if err != nil {
+		return err
+	}
```

Comment:
Please set the icon as `sha256:da834479c923584f4cbcdecc0dac61f32bef1d51e8aae598cf16bd154efab49f` if the icon is empty after parsing.
`sha256:da834479c923584f4cbcdecc0dac61f32bef1d51e8aae598cf16bd154efab49f` is the digest of the default icon introducing by PR #12683